lock-impl/src/main/java/com/palantir/lock/impl/LockServiceImpl.java

```diff
@@ -965,53 +973,40 @@ public LockServerOptions getLockServerOptions() {
         return options;
     }
 
-    private <T> List<T> queueToOrderedList(Queue<T> queue) {
-        List<T> list = Lists.newLinkedList();
-        while (!queue.isEmpty()) {
-            list.add(queue.poll());
-        }
-        for (T element : list) {
-            queue.add(element);
-        }
-        return list;
-    }
-
     /**
      * Prints the current state of the lock server to the logs. Useful for
      * debugging.
      */
     @Override
     public void logCurrentState() {
+        StringBuilder logString = getGeneralLockStats();
+        log.info("Current State: {}", logString.toString());
+
+        try {
+            logAllHeldAndOutstandingLocks();
+        } catch (IOException e) {
+            log.error("Can't dump state to Yaml: [{}]", e);
+            throw new IllegalStateException(e);
+        }
+    }
+
+    private void logAllHeldAndOutstandingLocks() throws IOException {
+        LockServiceStateLogger lockServiceStateLogger = new LockServiceStateLogger(heldLocksTokenMap, outstandingLockRequestMultimap);
+        lockServiceStateLogger.logLocks(lockStateLoggerDir);
+    }
+
+    private StringBuilder getGeneralLockStats() {
```

Comment:
Now that I'm thinking more about it, we can extract all of the metrics I can think of from the yaml itself so may not be necessary.  Could help usability a bit though.